var/spack/repos/builtin/packages/libmongoc/package.py

```diff
@@ -42,12 +42,24 @@ class Libmongoc(AutotoolsPackage):
     variant('snappy', default=True, description='Enable Snappy support.')
     variant('zlib', default=True, description='Enable zlib support.')
 
+    patch('snappy-zlib-libs.patch', when='@1.8.1')
```

Comment:
Could you download it from here: https://patch-diff.githubusercontent.com/raw/mongodb/mongo-c-driver/pull/466.patch